awesome_cart/templates/pages/cart.py

```diff
@@ -58,6 +58,24 @@ def get_context(context):
 	# remove? shipping is essential here anyways
 	context.shipping_enabled = 1 if settings.awc_shipping_enabled else 0
 
+	related_items = []
+	# build upsell sku list using or building cache if necessary
+	for item in awc_session.get("cart", {}).get("items", []):
+		item_related_items = awc.get_related_products_by_sku(
+			item.get("sku"), awc_session=awc_session, customer=customer)
+
+		# quick list deduping
+		related_items = related_items + list(set(item_related_items) - set(related_items))
```

Comment:
Also need to dedupe items already in cart

Reply:
Ah, yup. Fixing now